stdlib/AllocProfile/src/AllocProfile.jl

```diff
@@ -0,0 +1,110 @@
+module AllocProfile
+
+using Base.StackTraces
+using Base: InterpreterIP
+
+# raw results
+
+# modeled on jl_bt_element_t
+struct RawBacktraceElement
+    content::Csize_t
+end
+
+# matches RawBacktrace on the C side
+struct RawBacktrace
+    data::Ptr{RawBacktraceElement}
+    size::Csize_t
+end
+
+# matches RawAlloc on the C side
+struct RawAlloc
+    type::Ptr{Type}
+    backtrace::RawBacktrace
+    size::Csize_t
+end
+
+# matches RawAllocResults on the C side
+struct RawAllocResults
+    num_allocs::Csize_t
+    allocs::Ptr{RawAlloc}
+end
+
+function start(skip_every::Int=0)
+    ccall(:jl_start_alloc_profile, Cvoid, (Cint,), skip_every)
+end
+
+function stop()
+    raw_results = ccall(:jl_stop_alloc_profile, RawAllocResults, ())
+    decoded_results = decode(raw_results)
+    ccall(:jl_free_alloc_profile, Cvoid, ())
+    return decoded_results
+end
+
+# decoded results
+
+struct Alloc
+    # type::Type
+    type_addr::Ptr{Type} # TODO: fix segfault when loading this
+    stacktrace::Vector{StackTraces.StackFrame}
+    size::Int
+end
+
+function decode(raw_results::RawAllocResults)::Vector{Alloc}
+    out = Vector{Alloc}()
+    for i in 1:raw_results.num_allocs
+        raw_alloc = unsafe_load(raw_results.allocs, i)
+        push!(out, Alloc(
+            # unsafe_pointer_to_objref(convert(Ptr{Any}, raw_alloc.type)),
```

Comment:
this seems to work some of the time, but segfault some of the time (on pointers that look fine…)